src/main.ts

```diff
@@ -97,63 +102,22 @@ export class Minifier {
   }
 
   private checkReserved(str: string): boolean {
-    // From MDN's Lexical Grammar page
-    // (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar)
-    var reserved = [
-      'break',
-      'case',
-      'class',
-      'catch',
-      'const',
-      'continue',
-      'debugger',
-      'delete',
-      'do',
-      'else',
-      'export',
-      'extends',
-      'finally',
-      'for',
-      'function',
-      'if',
-      'import',
-      'in',
-      'instanceof',
-      'let',
-      'new',
-      'return',
-      'super',
-      'switch',
-      'this',
-      'throw',
-      'try',
-      'typeof',
-      'var',
-      'void',
-      'while',
-      'with',
-      'yield',
-      'enum',
-      'await',
-      'int',
-      'byte',
-      'char',
-      'goto',
-      'long',
-      'final',
-      'float',
-      'short',
-      'double',
-      'native',
-      'throws',
-      'boolean',
-      'abstract',
-      'volatile',
-      'transient',
-      'synchronized'
-    ];
+    if (!this.reservedJSKeywords) {
+      this.reservedJSKeywords = {};
```

Comment:
What about initializing this beforehand? For one, you have a constructor up there who's job it is to construct the class to be ready to use. But even better, make it a static field, and initialize it once when this code is loaded, e.g. by assigning the value from a static method (`static foo = bar()` where `bar` is a static method or function).